source/slang/slang-check-decl.cpp

```diff
@@ -7512,6 +7591,21 @@ namespace Slang
         }
         DeclAndCtorInfo structDeclInfo = DeclAndCtorInfo(m_astBuilder, this, structDecl, false);
 
+        // ensure all varDecl members are processed up to SemanticsBodyVisitor so we can be sure that if init expressions 
+        // of members are to be synthisised, they are.
+        bool isDefaultInitializableType = this->getShared()->findInheritance<InterfaceDecl>(structDecl, m_astBuilder->getDefaultInitializableType());
+        for (auto m : structDecl->members)
+        {
+            auto varDeclBase = as<VarDeclBase>(m);
+            if (!varDeclBase)
+                continue;
+            ensureDecl(m->getDefaultDeclRef(), DeclCheckState::DefaultConstructorReadyForUse);
+            if (!isDefaultInitializableType
```

Comment:
If !isDefaultInitializableType, we shouldn't enter this loop at all?

Reply:
we still need `ensureDecl(m->getDefaultDeclRef(), DeclCheckState::DefaultConstructorReadyForUse);` loop to handle member variables and setting their default inits (a struct without `isDefaultInitializableType` for example). This was a problem with the code the testing did not check for.

There shouldn't be a cost doing ensureDecl regardless -- we just do it earlier.